sim/firesim-lib/src/main/scala/configs/FireAxeTargetConfigs.scala

```diff
@@ -0,0 +1,145 @@
+  new RocketTileF1Config)
+// DOC include end: F1 Rocket Partition
+
+//////////////////////////////////////////////////////////////////////////////
+// - Xilinx U250 partition a RocketTile out
+// - Connect FPGAs using QSFP cables
+//
+// FPGA 0 (RocketTile) ----------- FPGA 1 (SoC subsystem)
+//////////////////////////////////////////////////////////////////////////////
+class RocketTileQSFPXilinxAlveoConfig extends Config(
+  new WithQSFP ++ // Use QSFP FPGA to FPGA connections. This will instantiate Aurora IPs in the FPGA shell.
+  new WithPartitionGlobalInfo(Seq(Seq("RocketTile"))) ++
+  new BaseXilinxAlveoU250Config // Xilinx U250 config
+)
+
+class RocketTileQSFPBase extends Config(
+  new WithPartitionBase ++
+  new RocketTileQSFPXilinxAlveoConfig)
+
+class RocketTileQSFPPartition0 extends Config(
+  new WithPartitionIndex(0) ++
+  new RocketTileQSFPXilinxAlveoConfig)
+
+//////////////////////////////////////////////////////////////////////////////
+// - Xilinx U250 partition RocketTiles onto separate FPGAs
+// - Connect FPGAs using QSFP cables
+//
+// FPGA 0 (RocketTile)     FPGA 1 (RocketTile_1)
+//             \            /
+//              \          /
+//           FPGA 2 (base SoC)
+//////////////////////////////////////////////////////////////////////////////
+class DualRocketTileQSFPXilinxAlveoConfig extends Config(
+  new WithQSFP ++
+  new WithPartitionGlobalInfo(Seq(
+    Seq("RocketTile"),
+    Seq("RocketTile_1"),
+  )) ++
+  new BaseXilinxAlveoU250Config)
+
+class DualRocketTileQSFPBase extends Config(
+  new WithPartitionBase ++
+  new DualRocketTileQSFPXilinxAlveoConfig)
+
+class DualRocketTileQSFP0 extends Config(
+  new WithPartitionIndex(0) ++
+  new DualRocketTileQSFPXilinxAlveoConfig)
+
+class DualRocketTileQSFP1 extends Config(
+  new WithPartitionIndex(1) ++
+  new DualRocketTileQSFPXilinxAlveoConfig)
+
+// DOC include start: Xilinx U250 Ring NoC Partition
+//////////////////////////////////////////////////////////////////////////////
+// - Xilinx U250 partition a ring NoC onto 3 FPGA connected as a ring
+//
+//        FPGA 0     ------------------ FPGA 1
+// - router 0 & tile 0               - router 2 & tile 2
+// - router 1 & tile 1               - router 3 & tile 3
+//                \                     /
+//                 \                   /
+//                  \                 /
+//                   \               /
+//                         FPGA 2
+//                  - router nodes 4 ~ 10
+//                  - SoC subsystem
+//////////////////////////////////////////////////////////////////////////////
+class QuadTileRingNoCTopoQSFPXilinxAlveoConfig extends Config(
+  new WithQSFP ++ // FireSim will instantiate Aurora IPs
+  new WithFireAxeNoCPart ++ // Compiler will detect NoC router nodes to partition
+  new WithPartitionGlobalInfo(Seq(
+    Seq("0", "1"),
+    Seq("2", "3"),
+    // base group has to be put last
+    (4 until 10).map(i => s"${i}")
+  )) ++
```

Comment:
Why is this API different than the other one where the base partition is implicitly everything else that isn't given in this Seq of Seqs. Why can't you just put:

```scala
  new WithPartitionGlobalInfo(Seq(
    Seq("0", "1"),
    Seq("2", "3")
  )) ++
```

Reply:
Good catch. This is to make the pass writing easier for the NoC partitioning pass. When you are removing/extracting individual modules out, that is all you need to know about the partitioning scheme. However, when you are partitioning across NoC router boundaries, you need a way of identifying all the router nodes that are irrelevant to the current partition as well. In theory, you can search for those router node in the FIRRTL pass, but that is just extra engineering complexity for not much gains.